server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ResilientClusterProvider.java

```diff
@@ -20,85 +20,53 @@
 import com.datastax.driver.core.exceptions.NoHostAvailableException;
-import com.github.steveash.guavate.Guavate;
 import com.nurkiewicz.asyncretry.AsyncRetryExecutor;
 import com.nurkiewicz.asyncretry.function.RetryCallable;
 
 @Singleton
 public class ResilientClusterProvider implements Provider<Cluster> {
-
-    public static final String CASSANDRA_KEYSPACE = "cassandra.keyspace";
-    private static final int DEFAULT_CONNECTION_MAX_RETRIES = 10;
-    private static final int DEFAULT_CONNECTION_MIN_DELAY = 5000;
-    private static final long CASSANDRA_HIGHEST_TRACKABLE_LATENCY_MILLIS = TimeUnit.SECONDS.toMillis(10);
-    private static final int DEFAULT_REPLICATION_FACTOR = 1;
-    private static final String DEFAULT_KEYSPACE = "apache_james";
-    private static final int DEFAULT_READ_TIMEOUT_MILLIS = 5000;
-    private static final int DEFAULT_CONNECT_TIMEOUT_MILLIS = 5000;
-    private static final String CASSANDRA_NODES = "cassandra.nodes";
-
     private static final Logger LOGGER = LoggerFactory.getLogger(ResilientClusterProvider.class);
+
     private final Cluster cluster;
 
     @Inject
-    private ResilientClusterProvider(CassandraSessionConfiguration cassandraSessionConfiguration, AsyncRetryExecutor executor) throws ConfigurationException, ExecutionException, InterruptedException {
-        PropertiesConfiguration configuration = cassandraSessionConfiguration.getConfiguration();
-        List<Host> servers = listCassandraServers(configuration);
-
-        QueryLoggerConfiguration queryLoggerConfiguration = getCassandraQueryLoggerConf(configuration);
-
-        int maxRetries = configuration.getInt("cassandra.retryConnection.maxRetries", DEFAULT_CONNECTION_MAX_RETRIES);
-        int minDelay = configuration.getInt("cassandra.retryConnection.minDelay", DEFAULT_CONNECTION_MIN_DELAY);
-
-        cluster = RetryExecutorUtil.retryOnExceptions(executor, maxRetries, minDelay, NoHostAvailableException.class)
-                .getWithRetry(getClusterRetryCallable(configuration, servers, queryLoggerConfiguration))
+    private ResilientClusterProvider(ClusterConfiguration configuration, AsyncRetryExecutor executor) throws ExecutionException, InterruptedException {
+        cluster = RetryExecutorUtil.retryOnExceptions(executor, configuration.getMaxRetry(), configuration.getMinDelay(), NoHostAvailableException.class)
+                .getWithRetry(getClusterRetryCallable(configuration))
                 .get();
     }
 
-    private RetryCallable<Cluster> getClusterRetryCallable(PropertiesConfiguration configuration, List<Host> servers, QueryLoggerConfiguration queryLoggerConfiguration) {
-        LOGGER.info("Trying to connect to Cassandra service at {} (list {})", LocalDateTime.now(), servers);
+    private RetryCallable<Cluster> getClusterRetryCallable(ClusterConfiguration configuration) {
+        LOGGER.info("Trying to connect to Cassandra service at {} (list {})", LocalDateTime.now(), configuration.getHosts());
```

Comment:
implicit toString?

Reply:
Was there before.

Anyway I'll fix it.